src/main/java/uk/gov/hmcts/reform/blobrouter/services/SasTokenGeneratorService.java

```diff
@@ -58,9 +59,13 @@ private BlobServiceSasSignatureValues getBlobServiceSasSignatureValues(String se
     }
 
     private StorageConfig getConfigForService(String serviceName) {
-        if (!serviceConfiguration.getStorageConfig().containsKey(serviceName)) {
+        StorageConfig config = serviceConfiguration.getStorageConfig().get(serviceName);
+        if (config == null) {
             throw new ServiceConfigNotFoundException("No service configuration found for " + serviceName);
+        } else if (!config.isEnabled()) {
+            throw new ServiceDisabledException("Service " + serviceName + " has been disabled.");
+        } else {
+            return config;
         }
```

Comment:
Let us understand this better: previous [commit](https://github.com/hmcts/blob-router-service/pull/30/commits/95a2cf79cf0bddc65e77686dd4c2d94b167b4bfb) it was okay to use `Optional` wrapper together with it's functional capabilities but after introducing another exception we are forcing ourselves to use if/else combination and _hardcoding_ `null` value.

Why not trusting java and use it as library properly plus it will give `+` points from future java releases (and currently used - 11).

```java
StorageConfig config = serviceConfiguration
    .getStorageConfig()
    .computeIfAbsent(serviceName, service -> {
        throw new ServiceConfigNotFoundException("No service configuration found for " + service);
    });

if (config.isDisabled()) {
    throw new ServiceDisabledException("Service " + serviceName + " has been disabled.");
}

return config;
```

Tiny note: at the moment we only need `isDisabled()` function 🙂

Reply:
I much prefer my version